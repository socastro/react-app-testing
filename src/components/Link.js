import {useState} from 'react';

const linkDesign = {
  hovered: 'hovered',
  normal: 'normal',
};

function Link({page, children}) {
  const [status, setStatus] = useState(linkDesign.normal);

  const onMouseEnter = () => {
    setStatus(linkDesign.hovered);
  };

  const onMouseLeave = () => {
    setStatus(linkDesign.normal);
  };

  return (
    <a
      className={status}
      href={page || '#'}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
}export default Link;