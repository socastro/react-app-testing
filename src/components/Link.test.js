'use strict';

import renderer from 'react-test-renderer';
import Link from './Link';

it('renderiza bien', () => {
  const tree = renderer
    .create(<Link page="http://www.youtube.com">APPWEB</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('cambia la clase cuando el mouse pasa encima', () => {
  const component = renderer.create(
    <Link page="http://www.youtube.com">APPWEB</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

//ejecuta manualmente el callback
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  // re-renderiza
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});