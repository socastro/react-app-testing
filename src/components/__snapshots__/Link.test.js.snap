// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cambia la clase cuando el mouse pasa encima 1`] = `
<a
  className="normal"
  href="http://www.youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  APPWEB
</a>
`;

exports[`cambia la clase cuando el mouse pasa encima 2`] = `
<a
  className="hovered"
  href="http://www.youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  APPWEB
</a>
`;

exports[`cambia la clase cuando el mouse pasa encima 3`] = `
<a
  className="normal"
  href="http://www.youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  APPWEB
</a>
`;

exports[`renderiza bien 1`] = `
<a
  className="normal"
  href="http://www.youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  APPWEB
</a>
`;
